src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -43,127 +85,725 @@ ActorInit En_Honotrap_InitVars = {
+        flameElem->unkC *= (0.006f * (((1.0f - flameGroup->unk4) * 0.8f) + 0.2f));
+        flameElem->flameScroll += flameScrollDisplacement;
+        flameElem->flameScroll %= 0x200u;
+    }
+
+    if (sp78 || (this->timer <= 0)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    temp_fs0 = flameGroup->unk0 * 120.0f;
+    temp_fs1 = flameGroup->unk4 * 120.0f;
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp68, &GET_PLAYER(play)->actor.world.pos);
+
+    if (sp68.z < temp_fs1) {
+        sp68.z = temp_fs1;
+    } else if (temp_fs0 < sp68.z) {
+        sp68.z = temp_fs0;
+    }
+
+    var_fs0_2 = Math_SinS((s16)(s32)(sp68.z * 212.3852f)) * 1.6f;
+    if (var_fs0_2 > 1.0f) {
+        var_fs0_2 = 1.0f;
+    }
+    sp80 *= ((1.0f - flameGroup->unk4) * 0.8f) + 0.2f;
+    if (sp80 > 0.2f) {
+        this->collider.cyl.dim.pos.x =
+            (s16)(s32)((Math_SinS(this->actor.shape.rot.y) * sp68.z) + this->actor.world.pos.x);
+        this->collider.cyl.dim.pos.y = (s16)(s32)(this->actor.world.pos.y - (24.0f * var_fs0_2));
+        this->collider.cyl.dim.pos.z =
+            (s16)(s32)((Math_CosS(this->actor.shape.rot.y) * sp68.z) + this->actor.world.pos.z);
+        this->collider.cyl.dim.radius = (s16)(s32)(15.0f * var_fs0_2);
+        this->collider.cyl.dim.height = (s16)(s32)(37.5f * var_fs0_2);
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.tris.base);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.tris.base);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.tris.base);
+    }
+}
+
+void EnHonotrap_Update(Actor* thisx, PlayState* play) {
+    static Vec3f sVelocity = { 0.0f, 0.0f, 0.0f };
+    static Vec3f sAccel = { 0.0f, 0.1f, 0.0f };
+    EnHonotrap* this = (EnHonotrap*)thisx;
+
+    if (this->timer > 0) {
+        this->timer--;
+    }
+    if ((this->actor.child != NULL) && (this->actor.child->update == NULL)) {
+        this->actor.child = NULL;
+    }
+    this->actionFunc(this, play);
+    if (this->collider.tris.base.acFlags & AC_HIT) {
+        EffectSsBomb2_SpawnLayered(play, &this->actor.world.pos, &sVelocity, &sAccel, 15, 8);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if (this->eyeState < HONOTRAP_EYE_SHUT) {
+        this->collider.tris.base.acFlags &= ~AC_HIT;
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.tris.base);
+    }
+}
+
+void EnHonotrap_UpdateFlame(Actor* thisx, PlayState* play) {
+    EnHonotrap* this = (EnHonotrap*)thisx;
+
+    if (this->timer > 0) {
+        this->timer--;
+    }
+    this->colChkFlags = 0;
+    this->bobPhase += 0x640;
+    if (this->actor.params != HONOTRAP_TYPE_FLAME_GROUP) {
+        this->actor.shape.yOffset = Math_SinS(this->bobPhase) * 1000.0f + 600.0f;
+    }
+    Actor_SetFocus(&this->actor, 5.0f);
+    Actor_PlaySfx(&this->actor, NA_SE_EV_BURN_OUT - SFX_FLAG);
+    this->actionFunc(this, play);
+    this->flameScroll -= 20;
+    this->flameScroll %= 0x200u;
```

Comment:
```suggestion
    this->flameScroll %= 0x200U;
```
Minor thing, but we've always done capital U when we make a number unsigned